src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -93,7 +119,7 @@ struct CheckMissedTypes {
       : policy_app_types_(policy_app_types), log_(log) {}
 
   bool operator()(const smart_objects::SmartArray::value_type &value) {
-    std::string app_type_str = value.asString();
+    std::string app_type_str = AppHMITypeToString((mobile_apis::AppHMIType::eType)value.asUInt());
```

Comment:
@jacobkeeler  please avoid using C-cast. Also value can be INVALID_ENUM, that is -1. So using it as unsigned int is not correct.